### PR TITLE
maria sparre project survey

### PR DESCRIPTION
**Pull request by MarySparrow**
Link to my deployed project:  https://optimistic-stonebraker-71d34e.netlify.app

My reflections on how this week's project turned out: 

Not so well as I hoped, did not understand the way to combine the components, but still really like React for how it simplifies things, just need to get how to :) 

Things I'd like to have clarified or explained in more detail:
How can I separate the Form into components?
how to show one question at the time, best practice to do it?
Please help me understand the things I have commented in the code 
-How do I align the radio buttons with text?

### CODE REVIEW

File: code/README.md [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
+
+## View it live
+
+Link to live site: 
+https://optimistic-stonebraker-71d34e.netlify.app
+
+Feedback 
+
+- How can I separate the Form into components? 
+- how to show one question at the time, best practice to do it?
```
Comment (Irina-web-dev) [inline code/README.md:22]: I did it like this:
     1. Define a state to update the current question: const [question, setQuestion] = useState(0)
     2. Function that will update the state:   const onNextQuestionChange = () => {
                                                                         setQuestion(question + 1)
                                                                     }
     3. Create a button that will call this function each time it is clicked: 
    <button onClick={onNextQuestionChange}> Next <button>
     4. Use the conditional for each question: 
     {question === 0 && (
<div className="container-input-fields">
        <div className="name-label">
          <label htmlFor={name}>Number of tickets:</label>
          <input 
            className="name-input" 
            id="name" 
            type="number" 
            value={name} 
            onChange={onNameChange}
          /> 
        </div>
          <button onClick={onNextQuestionChange}> Next <button>)
}

**Review by Irina-web-dev — CHANGES_REQUESTED** [review]
This is a very nice and clean website with easy to understand code behind it. Good job Maria! We do not always have to over style a website for it to look nice. Sometimes less is more ✨

As you have already mentioned some things needs to fixed like making radio buttons and text on the same line, creating at least one more component just to get the understanding on how to work with components and pass props. I have tried to give some advise on these questions, hope it will help you.

How do you think maybe at least media query for a mobile view would make your app more responsive and look nice on the small mobile device as well?

Have a great day 🦊
Irina

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
It is a great project, but please fix these radio buttons issue :)

File: code/src/index.css [hunk 2]
```diff
@@ -6,8 +6,91 @@ body {
...
+input[type=number], select {
+  width: 100%;
+  padding: 10px 20px;
+  margin: 20px 0;
+  display: inline-block;
+  border-radius: 5px;
+  box-sizing: border-box;
+}
+input[type=radio], select {
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (MarySparrow) [reply code/src/index.css:36]: Great feedback Irina, I have made it work with your help and some padding and removed margins :) good stuff!

File: code/src/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          <input 
+            id={date}
+            name="date" 
+            type="radio" 
+            value={date} 
+            required
+            onChange={onRadioChange}
+            checked={radio === date}>
+          </input>
+          {date}   {/*why is this needed?? */}
```
[the remark this answers is not in the document]
Comment (MarySparrow) [reply code/src/components/Form.js:122]: Hello, this I actually know, the confusion is why it is needed there in that position. It makes no sense to me. between two tags? The `{date }` is already mentioned in another place so why again... any input there to understand this better? ⭐

**Review by KarolCislowski — APPROVED** [review]
Great work 🥳

**Pull request closed without merging**